src/main/java/net/openhft/chronicle/core/io/TracingReferenceCounted.java

```diff
@@ -142,18 +140,10 @@ public void releaseLast(ReferenceOwner id) throws IllegalStateException {
         if (references.size() <= 1) {
             release(id);
         } else {
-            Exception e0 = null;
-            try {
-                release(id);
-            } catch (Exception e) {
-                e0 = e;
-            }
```

Comment:
This behaviour I think is incorrect, it would perform the release regardless of whether the reference was the last, this differed from the behaviour in VanillaReferenceCounted where the count was only decremented if it was <= 1

Reply:
When you check which reference still holds on to the resource,  you would have two stack traces of owners not released when one of them was actually attempted. i.e. isn't the source of an issue. This is likely to be more confusing than useful.

Reply:
Good point, I'll change Vanilla & the contract test to match this behaviour instead

Reply:
We don't have a test to show the expected message when the resources check fails.

Reply:
Sorry do you mean to add a test for the specific messages in the exceptions thrown by `release`? I think if we care that much about specific exceptions being thrown we should make specific exception sub-classes rather than testing message text.

There are tests for the specific messages that are logged when reservedBy returns false see [here](https://github.com/OpenHFT/Chronicle-Core/pull/446/files#diff-0149803deb8141c24d380ac84a76e201ca92dfd436d69f44d028ca1012302002R48) and [here](https://github.com/OpenHFT/Chronicle-Core/pull/446/files#diff-0149803deb8141c24d380ac84a76e201ca92dfd436d69f44d028ca1012302002R57)

Reply:
I added [some more tests around the specific errors thrown when releaseLast fails](https://github.com/OpenHFT/Chronicle-Core/pull/446/commits/da82b5b5f9312cb053ad993e951c2c807ca557c3)

Reply:
I can't see the commits, but I believe you